app/controllers/v0/message_drafts_controller.rb

```diff
@@ -4,36 +4,40 @@ class MessageDraftsController < SMController
     def create
       draft = MessageDraft.new(draft_params)
       raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
-      response = client.post_create_message_draft(draft_params)
 
-      render json: response,
+      draft_response = client.post_create_message_draft(draft_params)
+      render json: draft_response,
              serializer: MessageSerializer,
              status: :created
     end
 
     def update
-      draft = MessageDraft.new(draft_params)
-      raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
+      draft = client.post_create_message_draft(draft_params.merge(id: params[:id]))
 
-      client.post_create_message_draft(draft_params.merge(id: params[:id]))
+      # draft will fail validation if its a reply draft.
+      raise Common::Exceptions::ValidationErrors, draft unless draft.valid?
```

Comment:
shouldn't it have already failed validation when you did the client.post_create_message_draft?